doc/manual/src/design/overview.md

```diff
@@ -2,14 +2,21 @@
 
 Nix consists of layers that operate fairly independently.
 
-At the top is the *command line interface*.
+At the top is the *command line interface*, translating from invocations of Nix executables to interactions with the underlying layers.
 
-Below that is the Nix *expression language*, in which packages and configurations are written.
+Below that is the *Nix language*, a [purely functional programming](https://en.m.wikipedia.org/wiki/Purely_functional_programming) language.
+It is used to compose expressions which ultimately evaluate to self-contained *build plans*, used to derive *build results* from referenced *build inputs*.
 
-The command line and expression language are what users interact with most.
+::: {.note}
+The Nix language itself does not have a notion of *packages* or *configurations*.
+As far as we are concerned here, the inputs and results of a derivation are just data.
+In practice this amounts to a set of files in a file system.
```

Comment:
https://github.com/nix-community/nix/pull/7#discussion_r855162840 We should still leave that here for peace of mind of absolute beginners, even if keeping the store data in a file system is an implementation detail.

Reply:
Yeah moving it from the store layer intro to this note makes me OK with it. The note is more casual, and specifically there to address confusion.

Reply:
Good compromise :)